permissions/modules/revokePermission.ps1

```diff
@@ -0,0 +1,168 @@
+                if (-not [string]::IsNullOrEmpty($streamReaderResponse)) {
+                    $httpErrorObj.ErrorDetails = $streamReaderResponse
+                }
+            }
+        }
+        try {
+            $errorDetailsObject = ($httpErrorObj.ErrorDetails | ConvertFrom-Json)
+            # Make sure to inspect the error result object and add only the error message as a FriendlyMessage.
+            # $httpErrorObj.FriendlyMessage = $errorDetailsObject.message
+            $httpErrorObj.FriendlyMessage = $httpErrorObj.ErrorDetails # Temporarily assignment
+        }
+        catch {
+            $httpErrorObj.FriendlyMessage = $httpErrorObj.ErrorDetails
+        }
+        Write-Output $httpErrorObj
+    }
+}
+#endregion
+
+# Begin
+try {
+    # Verify if [aRef] has a value
+    if ([string]::IsNullOrEmpty($($actionContext.References.Account))) {
+        throw 'The account reference could not be found'
+    }
+
+    # Requesting authorization token
+    $splatRequestToken = @{
+        Uri    = "$($config.BaseUrl)/cas/oauth/token"
+        Method = 'POST'
+        Body   = @{
+            client_id     = $($config.ClientID)
+            client_secret = $($config.ClientSecret)
+            scope         = 'scim'
+            grant_type    = 'client_credentials'
+        }
+    }
+    $responseAccessToken = Invoke-RestMethod @splatRequestToken -Verbose:$false
+
+    $headers = [System.Collections.Generic.Dictionary[string, string]]::new()
+    $headers.Add('Authorization', "Bearer $($responseAccessToken.access_token)")
+    $headers.Add('Accept', 'application/json; charset=utf-8')
+    $headers.Add('Content-Type', 'application/json')
+
+    # Requesting authorization token
+    $splatRequestToken = @{
+        Uri    = "$($config.BaseUrl)/cas/oauth/token"
+        Method = 'POST'
+        Body   = @{
+            client_id     = $($config.ClientID)
+            client_secret = $($config.ClientSecret)
+            scope         = 'scim'
+            grant_type    = 'client_credentials'
+        }
+    }
+    $responseAccessToken = Invoke-RestMethod @splatRequestToken -Verbose:$false
+    Write-Information "Verifying if a Ysis account for [$($personContext.Person.DisplayName)] exists"
+    try {
+        $splatParams = @{
+            Uri         = "$($config.BaseUrl)/gm/api/um/scim/v2/users/$($actionContext.References.Account)"
+            Headers     = $headers
+            ContentType = 'application/scim+json;charset=UTF-8'
+        }
+        $responseUser = Invoke-RestMethod @splatParams -Verbose:$false
+    }
+    catch {
+        if ($_.Exception.Response.StatusCode -eq 404) {
+            $outputContext.AuditLogs.Add([PSCustomObject]@{
+                    Action  = "GrantPermission"
+                    Message = "Unable to assign permission [$($actionContext.References.Permission.displayName)]. Ysis account for [$($person.DisplayName)] not found. Possibly deleted" # Todo error message
+                    IsError = $true
```

Comment:
When revoking a permission and the account is not found. I would suggest the script will not go in to a error.

Reply:
Copy/paste error, fixed and improved